Compute mean IC50

pva701 opened the pull request:
## Description

<!--
Describes the nature of your changes. If they are substantial, you should
further subdivide this into a section describing the problem you are solving and
another describing your solution.
-->

## Related issue(s)

<!--
- Short description of how the PR relates to the issue, including an issue link.
For example
- Fixed #100500 by adding lenses to exported items

Write 'None' if there are no related issues (which is discouraged).
-->

https://issues.serokell.io/issue/EDNA-119

## :white_check_mark: Checklist for your Pull Request

<!--
Ideally a PR has all of the checkmarks set.

If something in this list is irrelevant to your PR, you should still set this
checkmark indicating that you are sure it is dealt with (be that by irrelevance).

If you don't set a checkmark (e. g. don't add a test for new functionality),
you must be able to justify that.
-->

#### Related changes (conditional)

- Tests
  - [ ] If I added new functionality, I added tests covering it.
  - [ ] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

- Documentation
  - [ ] I checked whether I should update the docs and did so if necessary:
    - [README](../tree/master/README.md)

#### Stylistic guide (mandatory)

- [ ] My commits comply with [the policy used in Serokell](https://www.notion.so/serokell/Commit-and-PR-policy-4cf98e1a910a415d86b5f2491d9af1af).
- [ ] My code complies with the [style guide](../tree/master/docs/code-style.md).

pva701 committed:
Add IC50 mean for selected subexperiment

Also make chart height bigger.

gromakovsky commented on frontend/src/store/selectors.ts:
```diff
@@ -175,25 +176,39 @@ export const selectedSubExperimentsQuery = selector<SubExperimentWithMeasurement
-    );
-    if ("Left" in ic50Sum) return ic50Sum;
-    return { Right: ic50Sum.Right / subs.length };
+    return evalMeanIC50(get(selectedSubExperimentsMetaQuery).map(x => x.item.result));
```
The issue says:
> Somewhere in Dashboard we show mean IC50 for all primary sub-experiments of all experiments where at least one sub-experiment is selected. I think that's what they want, not mean for all selected sub-experiments.

Currently we calculate average for all sub-experiments shown on plot, but I think they want each selected experiment to have equal weight, even if multiple sub-experiments are selected. They say:
> Необходим расчет среднего значения IC50 при выборе нескольких экспериментов

It's formulated imprecisely, so maybe I am wrong. Maybe it's better to have the following logic: if multiple sub-experiments are chosen from one experiment, we calculate average for them and then we take average of all these "averages". So if you have `se1` and `se2` from one experiment and `se3` for another experiment, we can return `((se1 + se2) / 2 + se3) / 2`. I don't insist on doing that and I don't know what they want. If you think I am saying nonsense, feel free to ignore. In general please try to maximize usefulness of this value divided by complexity of implementing it.

pva701 replied:
>So if you have se1 and se2 from one experiment and se3 for another experiment, we can return ((se1 + se2) / 2 + se3) / 2

Ok, maybe it makes sense... it's not hard to implement

sashasashasasha151 commented on frontend/src/components/IC50Line/IC50Tooltip.tsx:
```diff
@@ -0,0 +1,67 @@
+
+export type IC50Value = ResultDto<number | number[]> | Loadable<ResultDto<number | number[]>>;
+
+interface IC50LineProps {
```
Tooltip props

sashasashasasha151 commented on frontend/src/components/IC50Line/IC50Tooltip.tsx:
```diff
@@ -0,0 +1,67 @@
+  className?: string;
+}
+
+export function IC50Tooltip({ ic50, className }: IC50LineProps): React.ReactElement {
```
You write Tooltip here 4 times, but they are same all time. Like you can just return from some function all information which you need for Tooltip and write only one Tooltip

pva701 replied:
nichego ne ponyal

pva701 committed:
Add mean ic50 when target and compound is selected

Also, refactor ic50 evaluatio a little bit.

gromakovsky committed:
Remove `erMeanIC50` from ExperimentsResp

Problem: `erMeanIC50` is not used by frontend and most likely won't
be used because now we compute it on frontend.

Solution: remove it from `ExperimentsResp`.
